wallet/src/actors/worker/methods.rs

```diff
@@ -863,7 +869,7 @@ impl Worker {
 
     pub fn handle_block(
         &self,
-        block: types::ChainBlock,
+        block: Arc<types::ChainBlock>,
```

Comment:
If `index_block` takes a reference, so does this method:

```suggestion
        block: &types::ChainBlock,
```